### DISCUSSION TITLE
How to export ssh gpg keys from github to my local computer?

### DISCUSSION THREAD

**Discussion by jiapei100** [community/community#22311] [New to GitHub]
<p>It happened to me that my local laptop needs a fresh installation, and I didn’t do a good backup. I seriously do <strong>NOT</strong> want to generate ssh/gpg keys and have the key renewed on <a href="http://github.com" rel="nofollow noopener">github.com</a> again. I wonder if there is a way to just export the key on <a href="http://github.com" rel="nofollow noopener">github.com</a> to my local computer directly?</p>
<p>If so, how?</p>
<p>Thank you</p>
<p>Pei</p>

**Comment by seveas** [comment, marked as answer]
<p>Hi <a class="mention" href="/jiapei100">@jiapei100</a> ,</p>
<p>Unfortunately, that will not be possible. GItHub only has the public parts of your gpg and ssh keys, not the private parts, those only exist on your local computer. If you didn’t back those up before reinstalling, then I’m afraid that they are lost and you will need to generate new keys.</p>

**Comment by mpboom** [comment]
<p>However, if you are  <strong>100% sure</strong> the private key in question  <strong>is not</strong>   <strong>compromised</strong>   <strong>and safely destrcuted</strong> (which is  <strong>not</strong> the case by default with a fresh installation), you  <strong>can</strong> just leave the old key up on GitHub so your commits will continue to show up as  <strong>Verified</strong> while also putting your new key alongside the old key up on GitHub.</p>
